### PR TITLE
added Check constraint annotation

### LINKED ISSUE #1459: How add check and unique Constraints?
#### How to add check constraint and unique constraint?

### PR DESCRIPTION
**Pull request by davebehr1**
resolves #1459

Allow user to inject custom DDL into create Table for check constraints via annotations

### CODE REVIEW

File: dialect/entsql/annotation.go [hunk 1]
```diff
@@ -86,6 +86,14 @@ type Annotation struct {
 	//	}
 	//
 	OnDelete ReferenceOption `json:"on_delete,omitempty"`
+
+	// Check allows injecting custom "DDL" for setting the "CHECK" clause in "CREATE TABLE"
+	// table that has matching rows in the child table.
+	//	entsql.Annotation{
+	//		Check: < 10,
```
Comment (giautm) [inline dialect/entsql/annotation.go:93]: Is this line missing double-quotes?
```suggestion
	//		Check: "< 10",
```

File: dialect/entsql/annotation.go [hunk 2]
```diff
@@ -86,6 +86,14 @@ type Annotation struct {
 	//	}
 	//
 	OnDelete ReferenceOption `json:"on_delete,omitempty"`
+
+	// Check allows injecting custom "DDL" for setting the "CHECK" clause in "CREATE TABLE"
+	// table that has matching rows in the child table.
+	//	entsql.Annotation{
+	//		Check: "age < 10",
+	//	}
+	//
```
Comment (a8m) [inline dialect/entsql/annotation.go:95]:
```suggestion
	// Check allows injecting custom "DDL" for setting the "CHECK" clause in "CREATE TABLE".
	//
	//	entsql.Annotation{
	//		Check: "age < 10",
	//	}
	//
```

**Review by a8m — COMMENTED** [review]
Thanks for adding this @davebehr1.

**Comment by a8m** [comment]
Please add your Git email to your GitHub emails if you want to see your contributions in the repository.
I'll merge it after it.

**Comment by davebehr1** [comment]
> Please add your Git email to your GitHub emails if you want to see your contributions in the repository.
> I'll merge it after it.

I'm unsure what you mean by this @a8m. As far as I am aware I have included my email

File: entc/gen/template/migrate/schema.tmpl [hunk 3]
```diff
@@ -129,6 +129,9 @@ func init() {
 				{{- with $ant.Options }}
 					Options: "{{ . }}",
 				{{- end }}
+				{{- with $ant.Check }}
+					Check: "{{ . }}",
+				{{- end }}
```
Comment (a8m) [inline entc/gen/template/migrate/schema.tmpl:134]: This should be removed from the table's annotation and should be added to the column's annotation.

**Comment by a8m** [comment]
Thanks, @davebehr1! It was merged with https://github.com/ent/ent/pull/1519. 

@all-contributors please add @davebehr1 for code and @giautm for code review

**Comment by a8m** [comment]
@all-contributors please add @giautm for code review

**Pull request closed without merging**